crates/provider/src/provider/caller.rs

```diff
@@ -0,0 +1,124 @@
+use crate::ProviderCall;
+use alloy_json_rpc::{RpcParam, RpcReturn};
+use alloy_rpc_client::WeakClient;
+use alloy_rpc_types_eth::BlockId;
+use alloy_transport::{RpcError, Transport, TransportErrorKind, TransportResult};
+use std::borrow::Cow;
+
+/// A caller that helper convert `RpcWithBlock` and `EthCall` into a `ProviderCall`.
+pub trait Caller<T, Params, Resp>: Send + Sync
```

Comment:
so do we expect people to write custom `Caller` implementations which can customize `EthCall` and `WithBlock` behavior based on final `method` and `params`?

Reply:
Yes exactly!